planet/src/main/webapp/places.js

```diff
@@ -107,6 +107,32 @@ function updateSearch() {
         addPlaceDetails(places);
         map.fitBounds(bounds);
     });
+
+    // Add onclick function to option buttons
+    document.getElementById('hotel').onclick = function() {
+        document.getElementById('pac-input').value = 'hotel';
+        setSearchByButton(); 
+    }
+    document.getElementById('food').onclick = function() {
+        document.getElementById('pac-input').value = 'food';
+        setSearchByButton(); 
+    }
+    document.getElementById('tourist').onclick = function() {
+        document.getElementById('pac-input').value = 'tourist attractions';
+        setSearchByButton(); 
+    }
+    document.getElementById('nature').onclick = function() {
+        document.getElementById('pac-input').value = 'nature';
+        setSearchByButton(); 
+    }
+}
+
+/** Search map when option button is clicked */
+function setSearchByButton() {
+    const input = document.getElementById('pac-input');
+    google.maps.event.trigger(input, 'focus', {});
+    google.maps.event.trigger(input, 'keydown', { keyCode: 13 });
```

Comment:
what key is this? I'm guessing Enter?

Add a comment saying what these 3 lines are doing and why they're necessary